pkg/asset/machines/worker.go

```diff
@@ -0,0 +1,115 @@
+package machines
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/pkg/errors"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/machines/aws"
+	"github.com/openshift/installer/pkg/asset/machines/libvirt"
+	"github.com/openshift/installer/pkg/types"
+)
+
+func defaultAWSMachinePoolPlatform() types.AWSMachinePoolPlatform {
+	return types.AWSMachinePoolPlatform{
+		InstanceType: "m2.medium",
+	}
+}
+
+// Worker generates the machinesets for `worker` machine pool.
+type Worker struct {
+	MachineSetRaw     []byte
+	UserDataSecretRaw []byte
+}
+
+var _ asset.Asset = (*Worker)(nil)
+
+// Name returns a human friendly name for the Worker Asset.
+func (w *Worker) Name() string {
+	return "Worker Machines"
+}
+
+// Dependencies returns all of the dependencies directly needed by the
+// Worker asset
+func (w *Worker) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+		&machine.Worker{},
+	}
+}
+
+// Generate generates the Worker asset.
+func (w *Worker) Generate(dependencies asset.Parents) error {
+	installconfig := &installconfig.InstallConfig{}
+	wign := &machine.Worker{}
+	dependencies.Get(installconfig, wign)
+
+	var err error
+	w.UserDataSecretRaw, err = userData("worker-user-data", wign.File.Data)
+	if err != nil {
+		return errors.Wrap(err, "failed to create user-data secret for worker machines")
+	}
+
+	ic := installconfig.Config
+	pool := workerPool(ic.Machines)
+	numOfWorkers := int64(0)
+	if pool.Replicas != nil {
+		numOfWorkers = *pool.Replicas
+	}
+
+	switch ic.Platform.Name() {
+	case "aws":
+		config := aws.Config{
+			ClusterName: ic.ObjectMeta.Name,
+			Replicas:    numOfWorkers,
+			Region:      ic.Platform.AWS.Region,
+			Machine:     defaultAWSMachinePoolPlatform(),
+		}
+
+		tags := map[string]string{
+			"tectonicClusterID":                                         ic.ClusterID,
+			fmt.Sprintf("kubernetes.io/cluster/%s", ic.ObjectMeta.Name): "owned",
```

Comment:
Not sure what is more correct here, the actuator would need to ensure the tag is there, you shouldn't *need* to set this here, and it would be ideal if the actuator caught the duplicate. I'd just drop here and file a request for cloud team to check for dupes.